src/script/services/app-info.ts

```diff
@@ -97,6 +97,16 @@ export function setURL(url: string) {
   }
 }
 
+export function setCanonicalURL(initialURL) {
+  const parts = new URL(initialURL);
```

Comment:
will throw an exception if not a url, does that need to be caught?

Reply:
@lee-leonardo good find, this should always be a URL at this point as its not user generated, but I just pushed a commit to add a try / catch here.